src/gds/ua_record_datatype.h

```diff
@@ -0,0 +1,98 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ *
+ *    Copyright 2018 (c) Markus Karch, Fraunhofer IOSB
+ */
+
+#ifndef OPEN62541_UA_RECORD_DATATYPE_H
+#define OPEN62541_UA_RECORD_DATATYPE_H
+
+#include "ua_types.h"
+#include "ua_types_generated.h"
+
+/**
+ * ApplicationRecordDataType
+ * ^^^^^^^^^^^^^^^^^^^^^^^^^
+ */
+typedef struct {
+    UA_NodeId applicationId;
+    UA_String applicationUri;
+    UA_ApplicationType applicationType;
+    size_t applicationNamesSize;
+    UA_LocalizedText *applicationNames;
+    UA_String productUri;
+    size_t discoveryUrlsSize;
+    UA_String *discoveryUrls;
+    size_t serverCapabilitiesSize;
+    UA_String *serverCapabilities;
+} UA_ApplicationRecordDataType;
```

Comment:
Can't we generate this whole file from the `Opc.Ua.Gds.Types.bsd`

E.g. with the call (not tested):

```
# transport data types
ua_generate_datatypes(
    NAME "ua_gds"
    TARGET_SUFFIX "gds"
    NAMESPACE_IDX 0
    FILE_CSV "${PROJECT_SOURCE_DIR}/deps/ua-nodeset/GDS/OpcUaGdsModel.csv"
    FILES_BSD "${UA_FILE_TYPES_BSD}" "${PROJECT_SOURCE_DIR}/deps/ua-nodeset/GDS/Opc.Ua.Gds.Types.bsd"
    FILES_SELECTED "${PROJECT_SOURCE_DIR}/tools/schema/datatypes_transport.txt"
)
```

See also https://github.com/open62541/open62541/blob/6ad0222f619935347589f84eea08454ebcf22c0a/CMakeLists.txt#L748